### PR TITLE
Sql Queries

### PR DESCRIPTION
**Pull request by sebastienros**
Like lucene queries but with sql. Compared to Lucene, ad-hoc sql queries are not exposed with an Api, only for stored SQL queries.

Supports table prefix, parameter extraction. dialects (generates different sql for each database).

Parser is based on Irony. Parsing and converting would take less than 0.3ms on my machine.

### CODE REVIEW

File: src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                return View(model);
+            }
+
+            if (String.IsNullOrEmpty(model.Parameters))
+            {
+                model.Parameters = "{ }";
+            }
+            
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: I see no stop?

Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: Ellapsed, no need to stop.

Comment (Jetski5822) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: Thought you need to stop then get elapsed?

Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: Nope, this way you can display intermediary ellapsed time, while keeping the watch running. Think lap times.

File: src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        }
+
+        public Task<IActionResult> Query(string query, [FromServices] ITokenizer tokenizer)
+        {
+            query = String.IsNullOrWhiteSpace(query) ? "" : System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(query));
+            return Query(new AdminQueryViewModel { DecodedQuery = query }, tokenizer);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Query(AdminQueryViewModel model, [FromServices] ITokenizer tokenizer)
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: why are you passing the tokenizer when you have it injected in to the class?

File: src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+            if (String.IsNullOrEmpty(model.Parameters))
+            {
+                model.Parameters = "{ }";
+            }
+            
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            var connection = _store.Configuration.ConnectionFactory.CreateConnection();
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: this seems weird, kind of feel it sould be _store.CreateConntection()

File: src/OrchardCore.Modules/Orchard.Queries/Sql/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            services.AddScoped<IPermissionProvider, Permissions>();
+            services.AddScoped<IDisplayDriver<Query>, SqlQueryDisplayDriver>();
+            services.AddScoped<IQuerySource, SqlQuerySource>();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+        }
+
+        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+        }
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Startup.cs]: remove empty overload

File: src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs (hunk 3 above)
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.Queries/Sql/Controllers/AdminController.cs]: That's how YesSql is wire, you can customize the connection factory. Doesn't matter at all how.

**Pull request merged**